File: frontend/src/types/appointment.ts
/**
 * Appointment status.
 */
export const AppointmentStatus = {
  SCHEDULED: 'SCHEDULED',
  ONGOING: 'ONGOING',
  COMPLETED: 'COMPLETED',
  CANCELLED: 'CANCELLED'
} as const;

export type AppointmentStatus = typeof AppointmentStatus[keyof typeof AppointmentStatus];

/**
 * Appointment entity representing a scheduled appointment.
 */
export interface Appointment {
  id: string;
  patientId: string;
  dentistId: string;
  roomId: string;
  facilityId: string;
  startTime: string;
  endTime: string;
  status: AppointmentStatus;
  createdAt: string;
  updatedAt: string;
}

/**
 * Request DTO for creating an appointment.
 */
export interface CreateAppointmentRequest {
  patientId: string;
  dentistId: string;
  roomId: string;
  facilityId: string;
  startTime: string;
  endTime: string;
}

/**
 * Request DTO for updating an appointment.
 */
export interface UpdateAppointmentRequest {
  dentistId?: string;
  roomId?: string;
  startTime?: string;
  endTime?: string;
  status?: AppointmentStatus;
}

File: backend/src/main/java/com/smiles/appointments/domain/Appointment.java
package com.smiles.appointments.domain;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.UUID;

/**
 * Appointment entity representing a scheduled appointment.
 */
@Entity
@Table(name = "appointment")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Appointment {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "patient_id", nullable = false)
    private UUID patientId;

    @Column(name = "dentist_id", nullable = false)
    private UUID dentistId;

    @Column(name = "room_id", nullable = false)
    private UUID roomId;

    @Column(name = "facility_id", nullable = false)
    private UUID facilityId;

    @Column(name = "start_time", nullable = false)
    private Instant startTime;

    @Column(name = "end_time", nullable = false)
    private Instant endTime;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 50)
    private AppointmentStatus status;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private Instant updatedAt;
}
